src/Toolchain.ml

```diff
@@ -0,0 +1,615 @@
+module Caml = struct
+  module Array = Array
+end
+
+module R = Result
+open Bindings
+open Utils
+
+let pathMissingFromEnv = "'PATH' variable not found in the environment"
+
+let noPackageManagerFound =
+  {j| No package manager found. We support opam (https://opam.ocaml.org/) and esy (https://esy.sh/) |j}
```

Comment:
This error message appears a bit too often. If no package manager found, we should just gracefully downgrade the functionality we provide. There's quite a few OCaml projects out there that use neither opam nor esy, so we shouldn't annoy them with this message.

Unless there's a less annoying way to report this message, I would suggest that we remove it.

Reply:
I didn't realise that there are a significant number of projects out there using neither opam nor esy. Can we deal with this in a future PR? Until then I think we should explicitly show this message

Reply:
As discussed on slack, I'm afraid we can't delay this because it impacts the user experience for a large chunk of users.

Reply:
Fixed with `Global` variant introduced that looks for globally installed `ocamllsp`